k8s/configmap/configmap.go

```
@@ -1,89 +1,79 @@
-const optimisticMergeConflictError = "kubernetes api: Failure 409 Operation cannot be fulfilled on configmaps \"prometheus\": the object has been modified; please apply your changes to the latest version and try again"
-
-// Init just tries to get a K8s client created, and if it can't, bail
-func (c *ConfigMap) Init(ctx context.Context) {
-	cm := corev1.ConfigMap{}
-	// NewInClusterClient() creates a client that is forced into the same namespace
-	// as the entity in which the Client is created. So as long as bomb-squad runs
-	// in a container that resides in the same namespace as the Prometheus ConfigMap,
-	// we're good
-	client, err := k8s.NewInClusterClient()
-	if err != nil {
-		log.Fatal(err)
+// NewConfigMapWrapper returns a ConfigMapWrapper
+func NewConfigMapWrapper(client kubernetes.Interface, namespace string, configMapName string, dataKey string) *ConfigMapWrapper {
+	return &ConfigMapWrapper{
+		Client:  client.CoreV1().ConfigMaps(namespace),
+		Name:    configMapName,
+		DataKey: dataKey,
 	}
-	c.Client = client
+}
 
-	err = c.Client.Get(ctx, c.Client.Namespace, c.Name, &cm)
-	if err != nil {
-		log.Fatal(err)
-	}
-	c.CM = &cm
+// GetLocation implements github.com/Fresh-Tracks/bomb-squad/config.Configurator
+func (c *ConfigMapWrapper) GetLocation() string {
+	return c.DataKey
 }
 
-// ReadRawData pulls in value from the `data` key in the ConfigMap as-is
-func (c *ConfigMap) ReadRawData(ctx context.Context, key string) []byte {
-	cm := corev1.ConfigMap{}
-	err := c.Client.Get(ctx, c.Client.Namespace, c.Name, &cm)
+// Read implements github.com/Fresh-Tracks/bomb-squad/config.Configurator
+func (c *ConfigMapWrapper) Read() []byte {
+	dataKey := c.GetLocation()
+	cm, err := c.Client.Get(c.Name, v1.GetOptions{})
 	if err != nil {
-		log.Fatal(err)
-	}
-	c.CM = &cm
-
-	if res, ok := c.CM.Data[key]; !ok {
+		log.Printf("Failed to get ConfigMap in preparation for Configurator.Read(): %s\n", err)
```

Should return this error